primefaces-integration-tests/src/test/java/org/primefaces/integrationtests/datatable/DataTable026Test.java

```diff
@@ -0,0 +1,355 @@
+        page.roleFilter.deselect(Employee.Role.FINANCE.toString());
+        page.roleFilter.deselect(Employee.Role.HR.toString());
+        page.roleFilter.deselect(Employee.Role.QS.toString());
+
+        // Assert
+        List<Employee> employeesFiltered = employees.stream()
+                .filter(e -> e.getRole()==Employee.Role.DEVELOPER)
+                .collect(Collectors.toList());
+        assertEmployeeRows(dataTable, employeesFiltered);
+
+        // Act
+        page.roleFilter.select(Employee.Role.QS.toString(), true);
+
+        // Assert
+        employeesFiltered = employees.stream()
+                .filter(e -> e.getRole()==Employee.Role.DEVELOPER || e.getRole()==Employee.Role.QS)
+                .collect(Collectors.toList());
+        assertEmployeeRows(dataTable, employeesFiltered);
+
+        assertConfiguration(dataTable.getWidgetConfiguration());
+    }
+
+    @Test
+    @Order(98) //must be executed after all other tests because this one causes JS-errors
+    @DisplayName("DataTable: filter: range (part 1)")
+    public void testFilterRange1(Page page) {
+        // Arrange
+        DataTable dataTable = page.dataTable;
+
+        // Act
+        page.birthdateRangeFilter.getInput().sendKeys("1/1/1970 - 1/5/1970");
+        PrimeSelenium.guardAjax(page.birthdateRangeFilter.getInput()).sendKeys(Keys.TAB);
+
+        // Assert
+        List<Employee> employeesFiltered = employees.stream()
+                .filter(e -> e.getBirthDate().isAfter(LocalDate.of(1969, 12, 31)) && e.getBirthDate().isBefore(LocalDate.of(1970,1, 6)))
+                .collect(Collectors.toList());
+        assertEmployeeRows(dataTable, employeesFiltered);
+
+        // Setting range via Selenium (sendKeys) causes JS-errors. So we don´t check for them.
+//        assertConfiguration(dataTable.getWidgetConfiguration());
+    }
+
+    @Test
+    @Order(99) //must be executed after all other tests because this one causes JS-errors
+    @DisplayName("DataTable: filter: range (part 2)")
+    public void testFilterRange2(Page page) {
+        // Arrange
+        DataTable dataTable = page.dataTable;
+
+        // Act
+        page.birthdateRangeFilter.getInput().sendKeys("12/25/1969 - 1/3/1970");
+        PrimeSelenium.guardAjax(page.birthdateRangeFilter.getInput()).sendKeys(Keys.TAB);
+
+        // Assert
+        List<Employee> employeesFiltered = employeesFiltered = employees.stream()
+                .filter(e -> e.getBirthDate().isAfter(LocalDate.of(1969, 12, 24)) && e.getBirthDate().isBefore(LocalDate.of(1970,1, 4)))
+                .collect(Collectors.toList());
+        assertEmployeeRows(dataTable, employeesFiltered);
+
+        // Setting range via Selenium (sendKeys) causes JS-errors. So we don´t check for them.
+//        assertConfiguration(dataTable.getWidgetConfiguration());
+    }
+
+    private void assertConfiguration(JSONObject cfg) {
+        assertNoJavascriptErrors();
```

Comment:
Maybe we should add this to assertNoJavascriptErrors
https://stackoverflow.com/questions/51404360/how-to-clear-console-errors-using-selenium

Reply:
Yes we should add a `clearConsoleErrors()` method and add it.